lighthouse-core/test/report/v2/report-renderer-test.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* eslint-env mocha */
+/* global document self */
+
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+const jsdom = require('jsdom');
+const sampleResults = require('../../results/sample_v2.json');
+
+const TEMPLATES_FILE = fs.readFileSync(
+    path.join(__dirname, '../../../report/v2/templates.html'), 'utf8');
+
+function setupJsDomGlobals() {
+  global.document = jsdom.jsdom(TEMPLATES_FILE);
+  global.window = global.document.defaultView;
+  global.self = global.window;
+}
+
+function cleanupJsDomGlobals() {
+  global.document = undefined;
+  global.window = undefined;
+  global.self = undefined;
+}
+
+setupJsDomGlobals(); // Run jsdom setup code before report renderer is executed.
+require('../../../report/v2/report-renderer.js');
+
+describe('DOM', () => {
+  const window = self;
+  const DOM = window.DOM;
+
+  before(setupJsDomGlobals);
+  after(cleanupJsDomGlobals);
+
+  describe('setText', () => {
+    it('sets text of element', () => {
+      const el = document.createElement('div');
+      const result = DOM.setText(el, 'hello world');
+      assert.deepStrictEqual(result, el);
+      assert.equal(result.textContent, 'hello world');
+    });
+
+    it('does not fail when given null element argument', () => {
+      assert.doesNotThrow(() => {
```

Comment:
can drop the surrounding `assert.doesNotThrow` and just execute it. If it throws the test will fail